Bytecode failed verification 2 ?

I tried out the tiebreaking_expr with some dummyish data:

```from hail import *
from hail.representation import *
import subprocess
import os
hc = HailContext()

vds = hc.read('gs://daly_atgu_finnish_swedish_exomes_v8/daly_atgu_finnish_swedish_exomes_99prosAndPSC.vds')

vds = vds.filter_variants_expr('va.filters.isEmpty()', keep=True)

intervals = KeyTable.import_interval_list('gs://ibd-exomes/5k.txt')
vds = vds.filter_variants_table(intervals, keep=True)

vds = vds.variant_qc()
vds = vds.filter_variants_expr('va.qc.AF>0.01 && va.qc.callRate>0.99', keep=True)

table = hc.import_table('gs://daly_atgu_finnish_swedish_exomes_v8/finnibdPSC_isCase.txt', impute=True).key_by('s')
vds = vds.annotate_samples_table(table, root='sa.pheno')

vds = vds.repartition(100)
vds = vds.ibd_prune(0.35, tiebreaking_expr="if (sa1.pheno.iscase) 1 else 0")

vds.export_samples('gs://daly_atgu_finnish_swedish_exomes_v8/99pros_psc_relBelowPIHAT35.tsv','s=s')

```

But this results in an error:

```hail: info: Reading table to impute column types
hail: info: Finished type imputation
  Loading column `s' as type String (imputed)
  Loading column `case' as type Int (imputed)
Struct{
     pheno: Int
 }
[Stage 6:====================================================>(3347 + 1) / 3348]Verify Output for is/hail/codegen/generated/C3:
Traceback (most recent call last):
  File "/tmp/0a89b0df-1299-4db1-9e90-0efc77501684/99pros_psc_relatedness_short.py", line 26, in <module>
    vds = vds.ibd_prune(0.35, tiebreaking_expr="if (sa1.pheno) 1 else 0")
  File "<decorator-gen-322>", line 2, in ibd_prune
  File "/home/ec2-user/BuildAgent/work/179f3a9ad532f105/python/hail/java.py", line 112, in handle_py4j
hail.java.FatalError: IllegalStateException: Bytecode failed verification 2

Java stack trace:
java.lang.IllegalStateException: Bytecode failed verification 2
	at is.hail.asm4s.FunctionBuilder.classAsBytes(FunctionBuilder.scala:196)
	at is.hail.asm4s.FunctionBuilder.result(FunctionBuilder.scala:208)
	at is.hail.expr.CM.runWithDelayedValues(CM.scala:78)
	at is.hail.expr.Parser$.is$hail$expr$Parser$$evalNoTypeCheck(Parser.scala:49)
	at is.hail.expr.Parser$.is$hail$expr$Parser$$eval(Parser.scala:62)
	at is.hail.expr.Parser$.parseExpr(Parser.scala:77)
	at is.hail.expr.Parser$.parseTypedExpr(Parser.scala:87)
	at is.hail.methods.IBDPrune$$anonfun$16.apply(IBD.scala:308)
	at is.hail.methods.IBDPrune$$anonfun$16.apply(IBD.scala:308)
	at scala.Option.map(Option.scala:146)
	at is.hail.methods.IBDPrune$.apply(IBD.scala:308)
	at is.hail.variant.VariantDatasetFunctions$.ibdPrune$extension(VariantDataset.scala:529)
	at is.hail.variant.VariantDatasetFunctions.ibdPrune(VariantDataset.scala:526)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:237)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:280)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)

Hail version: 0.1-dfca956
Error summary: IllegalStateException: Bytecode failed verification 2
ERROR: (gcloud.dataproc.jobs.submit.pyspark) Job [0a89b0df-1299-4db1-9e90-0efc77501684] entered state [ERROR] while waiting for [DONE].```

Any idea what's wrong? Thanks!

--- a/src/main/scala/is/hail/expr/AST.scala
+++ b/src/main/scala/is/hail/expr/AST.scala
@@ -687,6 +687,9 @@ case class SymRef(posn: Position, symbol: String) extends AST(posn) {
 case class If(pos: Position, cond: AST, thenTree: AST, elseTree: AST)
   extends AST(pos, Array(cond, thenTree, elseTree)) {
   override def typecheckThis(ec: EvalContext): Type = {
+    if (cond.`type` != TBoolean) {
+      parseError(s"an `if` expression's condition must have type Boolean")
+    }
     (thenTree.`type`, elseTree.`type`) match {
       case (thenType, elseType) if thenType == elseType => thenType
       case (thenType: TNumeric, elseType: TNumeric) => TNumeric.promoteNumeric(Set(thenType, elseType))

--- a/src/test/scala/is/hail/methods/ExprSuite.scala
+++ b/src/test/scala/is/hail/methods/ExprSuite.scala
@@ -611,6 +611,16 @@ class ExprSuite extends SparkSuite {
         |else true
       """.stripMargin).contains(true))
 
+    val ifConditionNotBooleanMessage = "condition must have type Boolean"
+    TestUtils.interceptFatal(ifConditionNotBooleanMessage)(
+      eval[Int]("""if (1) 1 else 0"""))
+    TestUtils.interceptFatal(ifConditionNotBooleanMessage)(
+      eval[Int]("""if ("a") 1 else 0"""))
+    TestUtils.interceptFatal(ifConditionNotBooleanMessage)(
+      eval[Int]("""if (0.1) 1 else 0"""))
+    TestUtils.interceptFatal(ifConditionNotBooleanMessage)(
+      eval[Int]("""if ([1]) 1 else 0"""))
+
     assert(eval[Annotation]("""merge({a: 1, b: 2}, {c: false, d: true}) """).contains(Annotation(1, 2, false, true)))
     assert(eval[Annotation]("""merge(NA: Struct{a: Int, b: Int}, {c: false, d: true}) """).contains(Annotation(null, null, false, true)))
     assert(eval[Annotation]("""merge({a: 1, b: 2}, NA: Struct{c: Boolean, d: Boolean}) """).contains(Annotation(1, 2, null, null)))

